packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +621,117 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    // when transactionsController state changes
+    // check for pending transactions and start polling if there are any
+    this.subscribe(this.#checkForPendingTransactionAndStartPolling);
+
+    // TODO once v2 is merged make sure this only runs when
+    // selectedNetworkClientId changes
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      this.pendingTransactionTracker.onStateChange();
       this.onBootCleanup();
     });
 
     this.onBootCleanup();
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (_, patches) => {
+        if (this.enableMultichain) {
+          const networkClients = this.getNetworkClientRegistry();
+          patches.forEach(({ op, path }) => {
+            if (op === 'remove' && path[0] === 'networkConfigurations') {
+              const networkClientId = path[1] as NetworkClientId;
+              delete networkClients[networkClientId];
+            }
+          });
+
+          this.#refreshTrackingMap(networkClients);
```

Comment:
Very nice!

So for any network the user switches to, we create the necessary helpers which will still only start polling if there are pending transactions.

But incoming transactions are still explicitly requested by the public methods?

Reply:
I suppose it depends on what populates `getNetworkClientRegistry`.

Is it every used network per session? Or every saved client?

Reply:
> But incoming transactions are still explicitly requested by the public methods?

Right, should continue to be UI driven via `startIncomingTransactionPolling`, `stopIncomingTransactionPolling`

> Is it every used network per session? Or every saved client?
Every network client regardless of if it was used in that session or not